samples/drumthumper/src/main/java/com/google/oboe/sample/drumthumper/DrumThumperActivity.kt

```diff
@@ -142,23 +164,26 @@ class DrumThumperActivity : AppCompatActivity(), TriggerPad.DrumPadTriggerListen
             pad.addListener(this)
         }
 
-        mDrumPlayer.setupAudioStream()
-        mDrumPlayer.loadWavAssets(getAssets())
     }
 
     override fun onPause() {
         super.onPause()
-
-        mAudioMgr!!.unregisterAudioDeviceCallback(mDeviceListener)
     }
 
     override fun onStop() {
-        mDrumPlayer.teardownAudioStream()
         super.onStop()
+
```

Comment:
the onStop() sequence should be in reverse order to the onStart() sequence.
eg. call super.onStop() at the end.